src/filesys.cpp

```diff
@@ -715,20 +715,107 @@ std::string RemoveRelativePathComponents(std::string path)
 	pos = path.size();
 	while (pos != 0 && IsDirDelimiter(path[pos-1]))
 		pos--;
-	return path.substr(0, pos);
+	if (path != GetRootDir(path))
+		return path.substr(0, pos);
+	else
+		return path;
```

Comment:
If the path is root dir (/) w/o the change it would become empty and not working as expected as substr would remove the one and only directory separator.